packages/graphql/src/services/sessions.ts

```diff
@@ -50,6 +183,116 @@ export async function startSession(
   })
 }
 
+interface ActivateSessionBlockArgs {
+  sessionId: string
+  sessionBlockId: number
+}
+
+export async function activateSessionBlock(
+  { sessionId, sessionBlockId }: ActivateSessionBlockArgs,
+  ctx: ContextWithUser
+) {
+  const session = await ctx.prisma.session.findUnique({
+    where: {
+      id: sessionId,
+    },
+    include: { blocks: true },
+  })
+
+  if (!session || session.ownerId !== ctx.user.sub) return null
```

Comment:
Because of the efficiency of `findUnique`, this approach to verify ownership should be better than using `findFirst` and including the ownerId as a WHERE clause in the SQL statement. findFirst might cause a full scan of the table, while findUnique should use the index. At least I think so.